src/fidesops/ops/service/saas_request/override_implementations/firebase_auth_request_overrides.py

```diff
@@ -0,0 +1,155 @@
+                pds.append(
+                    {
+                        "display_name": pd.display_name,
+                        "provider_id": pd.provider_id,
+                        "email": pd.email,
+                    }
+                )
+            row["provider_data"] = pds
+        processed_data.append(row)
+
+    return processed_data
+
+
+@register("firebase_auth_user_update", [SaaSRequestType.UPDATE])
+def firebase_auth_user_update(
+    param_values_per_row: List[Dict[str, Any]],
+    policy: Policy,
+    privacy_request: PrivacyRequest,
+    secrets: Dict[str, Any],
+) -> int:
+    """
+    SaaS Request Override function used to integrate update requests with
+    Firebase Auth service. Specifically, it updates `User` records.
+    """
+    app = initialize_firebase(secrets)
+    rows_updated = 0
+    # each update_params dict correspond to a record that needs to be updated
+    for row_param_values in param_values_per_row:
+        email = row_param_values.get("email")
+        user: UserRecord = auth.get_user_by_email(email, app=app)
+        masked_fields = row_param_values["masked_object_fields"]
+        display_name = masked_fields.get("display_name", user.display_name)
+        phone_number = masked_fields.get("phone_number", user.phone_number)
+        disabled = masked_fields.get("disabled", user.disabled)
+        email_verified = masked_fields.get("email_verified", user.email_verified)
+
+        auth.update_user(
+            user.uid,
+            email=email,
+            display_name=display_name,
+            phone_number=phone_number,
+            disabled=disabled,
+            email_verified=email_verified,
+            app=app,
+        )
+        rows_updated += 1
+    return rows_updated
+
+
+@register("firebase_auth_user_delete", [SaaSRequestType.DELETE])
+def firebase_auth_user_delete(
+    param_values_per_row: List[Dict[str, Any]],
+    policy: Policy,
+    privacy_request: PrivacyRequest,
+    secrets: Dict[str, Any],
+) -> int:
+    """
+    SaaS Request Override function used to integrate delete requests with
+    Firebase Auth service. Specifically, it deletes `User` records.
+    """
+    app = initialize_firebase(secrets)
+    rows_updated = 0
+    # each update_params dict correspond to a record that needs to be updated
+    for row_param_values in param_values_per_row:
+        email = row_param_values.get("email")
+        user: UserRecord = auth.get_user_by_email(email, app=app)
+        auth.delete_user(user.uid, app=app)
+        rows_updated += 1
+    return rows_updated
+
+
+def initialize_firebase(secrets: Dict[str, Any]) -> App:
+    """
+    Initializes a Firebase "app" instance based on the given
+    secrets `dict`. The value of the `project_id` key of the
+    secrets dict is used for the `name` of the Firebase app.
+    """
+    creds = get_firebase_credentials(secrets)
+    project_id = creds.project_id
+    try:
+        app = firebase_admin.get_app(project_id)
```

Comment:
I remember you saying that we would get an error if we tried to initialize an "app" if one already existed. Is that why we're doing the try block? Not sure if I understand the issue but would Python's `global` keyword help here? We use it in `cache.py` to always return a single instance of the Redis cache.

Reply:
actually, this is sort of the opposite of what i'd described. after mentioning the piece about initializing an app if it already exists, i found this method in the SDK that allows us to effectively retrieve an app by name/id, if it already exists; if it doesn't exist, it throws an error. and so we catch that here and initialize if it doesn't exist yet.

the `global` keyword may be a bit trickier here, because we don't want to assume there's only a single firebase `app` per fidesops runtime - that would prevent a single fidesops instance from integrating with multiple firebase projects, which is a limitation we should avoid if possible imo. perhaps we could maintain a global map of firebase `app`s tied to particular saas config instances, or something of the sort -- but that's effectively what we're getting for free here with the admin SDK!

i don't love that we need to try/catch, but other than that, i actually find this a pretty clean implementation. let me know what you think.